organization.go

```diff
@@ -439,19 +447,53 @@ func (s *organizations) ReadRunQueue(ctx context.Context, organization string, o
 	return rq, nil
 }
 
-func (s *organizations) ReadDataRetentionPolicy(ctx context.Context, organization string) (*DataRetentionPolicy, error) {
+func (s *organizations) ReadDataRetentionPolicy(ctx context.Context, organization string) (*DataRetentionPolicyChoice, error) {
 	if !validStringID(&organization) {
 		return nil, ErrInvalidOrg
 	}
 
+	// The API to read the drp is org/<name>/relationships/data-retention-policy
+	// However, this API can return multiple "types" (e.g. data-retention-policy-delete-olders, or data-retention-policy-don-deletes)
+	// Ideally we would deserialize this directly into the choice type (DataRetentionPolicyChoice)...however, there isn't a way to
+	// tell the current jsonapi implementation that the direct result of an endpoint could be different types. Relationships can be polymorphic,
+	// but the direct result of an endpoint can't be (as far as the jsonapi implementation is concerned)
```

Comment:
This is potentially something we could improve in the jsonapi fork...it would be useful to specify that the direct result of an API (not just a relationship) is polymorphic, and should populate a choice type.

Im not sure exactly what that would look like...maybe we would also need an `UnmarshalPolymorphicPayload` function. Or we could add some sort of tag to the choice type (`DataRetentionPolicyChoice`) to indicate that it should be treated as polymorphic...something like:
```
type DataRetentionPolicyChoice struct {
       Type string `jsonapi:"primary,polymorphic"`
	DataRetentionPolicy            *DataRetentionPolicy
	DataRetentionPolicyDeleteOlder *DataRetentionPolicyDeleteOlder
	DataRetentionPolicyDontDelete  *DataRetentionPolicyDontDelete
}
```